src/Uno.UI/UI/Xaml/Media/Animation/Timeline.cs

```diff
@@ -164,13 +164,16 @@ internal BindingPath PropertyInfo
 		{
 			get
 			{
-				if (_propertyInfo == null)
+				// Don't use the cached _propertyInfo if TargetProperty has the changed.
+				// Target changes are handled by setting `DataContext` in `SetElementNameTarget`.
+				var path = Storyboard.GetTargetName(this);
+				if (_propertyInfo == null || _propertyInfo.Path != path)
```

Comment:
Another possible fix is to add a setter for `BindingPath.Path`, and update it whenever `SetTargetProperty` is called.

Reply:
That would make the internals of binding path a bit complex to manage for an edge case.